junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/EnumSource.java

```diff
@@ -121,7 +121,15 @@
 		 *
 		 * @see java.util.stream.Stream#anyMatch(java.util.function.Predicate)
 		 */
-		MATCH_ANY(Mode::validatePatterns, (name, patterns) -> patterns.stream().anyMatch(name::matches));
+		MATCH_ANY(Mode::validatePatterns, (name, patterns) -> patterns.stream().anyMatch(name::matches)),
+
+		/**
+		 * Select only those enum constants whose names don't match any pattern supplied
+		 * via the {@link EnumSource#names} attribute.
+		 *
+		 * @see java.util.stream.Stream#noneMatch(java.util.function.Predicate)
+		 */
+		MATCH_NONE(Mode::validatePatterns, (name, patterns) -> patterns.stream().noneMatch(name::matches));
```

Comment:
```suggestion
		@API(status = EXPERIMENTAL, since = "5.9")
		MATCH_NONE(Mode::validatePatterns, (name, patterns) -> patterns.stream().noneMatch(name::matches));
```

plus import